### PR TITLE
Add Deno webhook signing example

### PR DESCRIPTION
**Pull request by anniel-stripe**
## Summary
Add a Deno webhook signing example. Also updated our test script to actually exit with an error if the server never starts (the test didn't fail when Deno wasn't installed in CI).

### CODE REVIEW

File: examples/webhook-signing/deno/main.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+import {serveListener} from 'https://deno.land/std@0.116.0/http/server.ts';
+import 'https://deno.land/x/dotenv/load.ts';
+
+import Stripe from 'npm:stripe@^11.16';
+
+const stripe = Stripe(Deno.env.get('STRIPE_API_KEY'));
+
+// Must specify hostname explicitly, see https://github.com/denoland/deno/issues/5144
+const server = Deno.listen({hostname: '127.0.0.1', port: 8080});
```
Comment (pakrym-stripe) [inline examples/webhook-signing/deno/main.ts:11]: can we bind to port 0 (dynamic) and retrieve the actual value from the server?

File: examples/webhook-signing/test/main.ts [hunk 2]
```diff
@@ -112,7 +112,11 @@ const sendTestRequest = async (url: string): Promise<string> => {
     process.exit(0);
   } catch (e) {
     console.error(e);
+    throw e;
   } finally {
     server.kill();
   }
-})().catch(console.log);
+})().catch((e) => {
```
Comment (pakrym-stripe) [inline examples/webhook-signing/test/main.ts:119]: Was a failing test not failing the CI?

**Review by pakrym-stripe — CHANGES_REQUESTED** [review]
Can we get dynamic port binding working? It will make test less flaky and resistant to the situation where another process already uses the port.

File: examples/webhook-signing/test/main.ts (hunk 2 above)
Comment (anniel-stripe) [reply examples/webhook-signing/test/main.ts:119]: Before, if the server hadn't started (e.g., if main.ts wasn't executable or `deno` wasn't installed), then the test would still pass, for example in https://github.com/stripe/stripe-node/actions/runs/4696233056/jobs/8326148593 the test exits 1 with:
```
sh: 1: 
deno: not found
```
but CI does not fail.

File: examples/webhook-signing/deno/main.ts (hunk 1 above)
Comment (anniel-stripe) [reply examples/webhook-signing/deno/main.ts]: Yes! Added in latest commit

**Review by pakrym-stripe — APPROVED** [review]
Nice work!

**Pull request merged**